pkg/operator/controllers/autosizednodes/autosizednodes_controller.go

```diff
@@ -90,7 +94,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return ctrl.Result{}, fmt.Errorf("could not fetch KubeletConfig: %w", err)
 	}
 
-	// If already exists, update the spec
+	// If already exists, update the spec and annotations
+	if config.Annotations == nil {
```

Comment:
nit; but should be using `len (config.Annotations == 0)` to determine if the map is empty https://go.dev/ref/spec#Length_and_capacity